src/main/java/io/nats/client/ConnectionListener.java

```diff
@@ -20,31 +20,52 @@
 public interface ConnectionListener {
     public enum Events {
         /** The connection has successfully completed the handshake with the nats-server. */
-        CONNECTED("nats: connection opened"),
+        CONNECTED(true, "opened"),
         /** The connection is permanently closed, either by manual action or failed reconnects. */
-        CLOSED("nats: connection closed"),
+        CLOSED(true, "closed"),
         /** The connection lost its connection, but may try to reconnect if configured to. */
-        DISCONNECTED("nats: connection disconnected"), 
+        DISCONNECTED(true, "disconnected"),
         /** The connection was connected, lost its connection and successfully reconnected. */
-        RECONNECTED("nats: connection reconnected"), 
+        RECONNECTED(true, "reconnected"),
         /** The connection was reconnected and the server has been notified of all subscriptions. */
-        RESUBSCRIBED("nats: subscriptions re-established"),
+        RESUBSCRIBED(false, "subscriptions re-established"),
         /** The connection was told about new servers from, from the current server. */ 
-        DISCOVERED_SERVERS("nats: discovered servers"),
+        DISCOVERED_SERVERS(false, "discovered servers"),
         /** Server Sent a lame duck mode. */
-        LAME_DUCK("nats: lame duck mode");
+        LAME_DUCK(false, "lame duck mode");

```

Comment:
I modified this class for more granular information about the event and to allow for different types of reporting. It is backward compatible, meaning that the toString did not change text.